### PR TITLE
NET-5073 - ProxyConfiguration: implement various connection options

### PR DESCRIPTION
**Pull request by jmurret**
### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go [hunk 1]
```diff
@@ -384,20 +402,34 @@ func (b *Builder) addLocalAppCluster(clusterName string) *Builder {
...
+				cluster.GetEndpointGroup().GetStatic().GetConfig().CircuitBreakers = &pbproxystate.CircuitBreakers{
+					UpstreamLimits: &pbproxystate.UpstreamLimits{
+						MaxConnections: &wrapperspb.UInt32Value{Value: b.proxyCfg.DynamicConfig.InboundConnections.MaxInboundConnections},
+					},
+				}
+			}
+		}
+	}
+
+	b.proxyStateTemplate.ProxyState.Clusters[clusterName] = cluster
```
Comment (ishustava) [inline internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go:427]: I think we probably also need to add request timeout for l7 (I think this probably goes on the route)

File: internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go [hunk 2]
```diff
@@ -289,6 +298,15 @@ func (l *ListenerBuilder) addInboundRouter(clusterName string, routeName string,
 				AlpnProtocols: []string{getAlpnProtocolFromPortName(portName)},
 			},
 		}
+
+		if ic != nil {
+			// MaxInboundConnections is uint32 that is used on:
+			// - router destinations MaxInboundConnection (uint64)
+			// - cluster circuit breakers UpstreamLimits.MaxConnections (uint32)
+			// It is cast to a uint64 here similarly as it is to the proxystateconverter code
+			r.GetL4().MaxInboundConnections = uint64(ic.MaxInboundConnections)
```
[the remark this answers is not in the document]
Comment (jmurret) [reply internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go:307]: i am afraid of allowing uint64 for user configuration and then casting down to uint32 for UpstreamLimits.MaxConnections.

[the remark this answers is not in the document]
Comment (ishustava) [reply internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go:307]: Oh that's fair!

File: internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go [hunk 3]
```diff
@@ -264,10 +267,16 @@ func (b *Builder) addInboundListener(name string, workload *pbcatalog.Workload)
 	// Add TLS inspection capability to be able to parse ALPN and/or SNI information from inbound connections.
 	listener.Capabilities = append(listener.Capabilities, pbproxystate.Capability_CAPABILITY_L4_TLS_INSPECTION)
 
+	if b.proxyCfg != nil && b.proxyCfg.DynamicConfig != nil && b.proxyCfg.DynamicConfig.InboundConnections != nil {
```
[the remark this answers is not in the document]
Comment (jmurret) [reply internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go]: fixed in 0e39f20d8b9246487fb184ad281ade2c7c4396cc

File: internal/mesh/internal/controllers/sidecarproxy/builder/expose_paths.go [hunk 4]
```diff
@@ -22,7 +22,7 @@ func (b *Builder) buildExposePaths(workload *pbcatalog.Workload) {
 				buildListener()
 
 			b.addExposePathsRoute(exposePath, clusterName).
-				addLocalAppCluster(clusterName).
+				addLocalAppCluster(clusterName, fmt.Sprintf("%d", exposePath.LocalPathPort)).
```
[the remark this answers is not in the document]
Comment (jmurret) [reply internal/mesh/internal/controllers/sidecarproxy/builder/expose_paths.go]: great call. removed in 0e39f20d8b9246487fb184ad281ade2c7c4396cc

File: proto-public/pbmesh/v2beta1/connection.proto [hunk 5]
```diff
@@ -15,7 +15,7 @@ message ConnectionConfig {
 
 // Referenced by ProxyConfiguration
 message InboundConnectionsConfig {
-  uint64 max_inbound_connections = 12;
+  uint32 max_inbound_connections = 12;
   BalanceConnections balance_inbound_connections = 13;
```
[the remark this answers is not in the document]
Comment (jmurret) [reply proto-public/pbmesh/v2beta1/connection.proto]: changed and proto regenedated in 0e39f20d8b9246487fb184ad281ade2c7c4396cc

File: internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go (hunk 1 above)
Comment (jmurret) [reply internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go:434]: great call.  I originally had my golden file utilize an HTTP port and wound up changing this and losing track of this.  thank you.  this was fixed in 0e39f20d8b9246487fb184ad281ade2c7c4396cc

**Review by ishustava — APPROVED** [review]
🎉

**Pull request merged**